CMakeLists.txt

```diff
@@ -563,7 +567,7 @@ rocm_package_setup_component(profiler
 )
 add_subdirectory(profiler)
 
-if(GPU_TARGETS MATCHES "gfx9" OR GPU_ARCHS)
+if(DEFINED CK_USE_CODEGEN AND (GPU_TARGETS MATCHES "gfx9" OR GPU_ARCHS))
```

Comment:
This should not check if its defined as it will use the codegen even when the user explicitly disables it(ie `cmake -DCK_USE_CODEGEN=Off`). Instead just do `if(CK_USE_CODEGEN AND (GPU_TARGETS MATCHES "gfx9" OR GPU_ARCHS))`.